File: backend/app/models/capture.py
from datetime import datetime, timezone
from enum import Enum
from uuid import UUID, uuid4

from sqlmodel import Field, SQLModel


class CaptureType(str, Enum):
    TEXT = "text"
    LINK = "link"
    FILE = "file"
    PHOTO = "photo"
    VOICE = "voice"


class CaptureBase(SQLModel):
    title: str
    content: str
    capture_type: CaptureType = CaptureType.TEXT

    workspace_id: UUID | None = Field(
        default=None,
        foreign_key="workspace.id",
    )

    session_id: UUID | None = Field(
        default=None,
        foreign_key="session.id",
    )


class CaptureCreate(CaptureBase):
    """
    Data accepted when creating a capture.

    Both workspace and session context are optional.
    """

    pass


class Capture(CaptureBase, table=True):
    """
    Represents a quick thought, note, link, or future artifact.
    """

    id: UUID = Field(default_factory=uuid4, primary_key=True)

    created_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc)
    )

File: frontend/lib/models/capture.dart
class Capture {
  final String id;
  final String title;
  final String content;
  final String captureType;
  final String createdAt;
  final String? workspaceId;
  final String? sessionId;

  const Capture({
    required this.id,
    required this.title,
    required this.content,
    required this.captureType,
    required this.createdAt,
    this.workspaceId,
    this.sessionId,
  });

  factory Capture.fromJson(Map<String, dynamic> json) {
    return Capture(
      id: json['id'] as String,
      title: json['title'] as String,
      content: json['content'] as String,
      captureType: json['capture_type'] as String,
      createdAt: json['created_at'] as String,
      workspaceId: json['workspace_id'] as String?,
      sessionId: json['session_id'] as String?,
    );
  }
}